crates/flowctl/src/catalog/publish.rs

```diff
@@ -12,9 +12,12 @@ pub struct Publish {
     /// This flag is required if running flowctl non-interactively, such as in a shell script.
     #[clap(long)]
     auto_approve: bool,
-    /// Data-plane into which created specifications will be placed.
-    #[clap(long, default_value = "ops/dp/public/gcp-us-central1-c2")]
-    default_data_plane: String,
```

Comment:
There's docs in `site/docs/guides/flowctl/ci-cd.md` that refer to this argument, and will need updated.

May or may not be worth it, but we could also leave this argument here, but hidden, and print an error message if it gets passed. Just thinking about users upgrading, and wanting to make sure we communicate the breaking change effectively. Would also be good if we remember to include this in the release notes.

Reply:
Updated docs, and I added back `--default-data-plane` as an alias for `--init-data-plane`.